### ISSUE TITLE
Enum is not assignable to type string

### ISSUE THREAD

**Issue by SiegeSailor** [microsoft/TypeScript#43039]
# Bug Report

Strings in an Enum form are not assignable to fields that require type string.

### 🔎 Search Terms

- Enum as a string parameter
- Enum is not assignable to type string

### 🕗 Version & Regression Information

- This is the behavior between versions `4.1.15` and `4.2.2`.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wChSkA7AVzwGEJLNgBzOAb1Lm7gDkUQSOAF4CAFQDK+ADRceAdWAATGAAsRcAEwAGbbJ5wAEklar4ondtIBfckgAekWHCVJMKagBss1ShmCMcACi9gJgnkgAFACUnAZEMNRQlHAAPErAAG5wysLsDEysAHT8gtYAfKkA9BmZ5TZAA)

### 💻 Code

```TypeScript
import React from 'react'

enum Config {
    Name = 'TS',
    Width = 200,
    Height = 200
}

export default function Example(){
    return <div id={Config.Name}></div>
}
```

### 🙁 Actual behavior

The terminal is complaining with the following message:

```
Type 'Config' is not assignable to type 'string | undefined'.
```

### 🙂 Expected behavior

Code works without the error message.

**Comment by SiegeSailor** [comment]
### 📜 Update

 This bug happens to the number parameter as well.

```
Type 'Config' is not assignable to type 'number | undefined'.ts(2322)
```

**Comment by RyanCavanaugh** [comment]
Mixed-type enums aren't actually supported; you *should* get an error on `Name` there but our initialization rules are not tight enough.

**Comment by SiegeSailor** [comment]
It is common to use mixed-type enums, I suggest supporting it rather than reinforcing the initialization rule.

**Comment by lemoustachiste** [comment]
any clean workaround for this issue?

I have a set of enum properties that are helpful in my library, but essentially they are glorified strings (as highlighted by the above comment). When I am passing them down to a lower level library (over which I have also control, but cannot know of the enum), I just expect a string there. But I am now getting the error mentioned in this issue.

**Comment by MartinJohns** [comment]
Use a type assertion.
